diff --git a/net/ipv4/tcp_bbr.c b/net/ipv4/tcp_bbr.c
index e09389c378cc..c44fd8d1930f 100644
--- a/net/ipv4/tcp_bbr.c
+++ b/net/ipv4/tcp_bbr.c
@@ -482,7 +482,7 @@ static u32 bbr_tso_segs_generic(struct sock *sk, unsigned int mss_now,
 {
 	struct bbr *bbr = inet_csk_ca(sk);
 	u32 segs, r;
-	u64 bytes;
+	unsigned long bytes;
 
 	/* Budget a TSO/GSO burst size allowance based on bw (pacing_rate). */
 	bytes = READ_ONCE(sk->sk_pacing_rate) >> READ_ONCE(sk->sk_pacing_shift);
